bundles/org.eclipse.packagedrone.utils.rpm/src/org/eclipse/packagedrone/utils/rpm/build/RpmBuilder.java

```diff
@@ -754,9 +787,8 @@ private Path makeTargetFile ( final Path target )
 
     private String makeDefaultFileName ()
     {
-        final StringBuilder sb = new StringBuilder ( RpmLead.toLeadName ( this.name, this.version ) );
-        sb.append ( '-' ).append ( this.architecture ).append ( ".rpm" );
-        return sb.toString ();
+        assert this.rpmFileNameProvider != null;
```

Comment:
Adding an assert might be right, but Package Drone doesn't use them and I don't think it would add much value here. As you do get an unexpected exception anyway. It would be great if you could remove it.

Reply:
hmmm... I used the assert since I saw it in other places in the project (eg. BCJEncoder). 
I've removed that line.